pinot-plugins/pinot-minion-tasks/pinot-minion-builtin-tasks/src/main/java/org/apache/pinot/plugin/minion/tasks/mergerollup/MergeRollupTaskGenerator.java

```diff
@@ -339,6 +405,18 @@ public String getTaskType() {
           .info("Finished generating task configs for table: {} for task: {}, numTasks: {}", offlineTableName, taskType,
               pinotTaskConfigsForTable.size());
     }
+
+    // Reset watermarks for invalid tables
```

Comment:
I'm a little bit confused why we need to clean up metrics for invalid tables since we will not create metrics for these tables. Are we trying to handle schema evolution here?

Reply:
I was thinking of the case where we added the merge config and then removed.